include/rdma/rdma_cma.h

```diff
@@ -197,6 +197,29 @@ struct rdma_addrinfo {
 	struct rdma_addrinfo	*ai_next;
 };
 
+/* Multicast join compatibility mask attributes */
+enum rdma_cm_join_mc_attr_mask {
+	RDMA_CM_JOIN_MC_ATTR_ADDRESS	= 1 << 0,
+	RDMA_CM_JOIN_MC_ATTR_JOIN_FLAGS	= 1 << 1,
+	RDMA_CM_JOIN_MC_ATTR_RESERVED	= 1 << 2,
+};
+
+/* Multicast join flags */
+enum rdma_cm_mc_join_flags {
+        RDMA_MC_JOIN_FLAG_FULLMEMBER,
+        RDMA_MC_JOIN_FLAG_SENDONLY_FULLMEMBER,
+        RDMA_MC_JOIN_FLAG_RESERVED,
+};
```

Comment:
The rdma cm is intended to mimic a sockets-like API, not a native IB one.  The current code matches with setsockopt - IP_ADD_MEMBERSHIP.  There's no mapping for 'full member' for sockets.

Unlike sockets, we do have to join for send only transfers.  For that I'm suggesting that we add a flag, SENDONLY, that the app can set when joining.  If left unset, the QP is attached for send/receive.  This will also allow us to add other flags, like RDMA_MC_NO_LOOPBACK, to disable loopback traffic.

We don't need a kernel change.  We should be able to map what the user provides to whatever the kernel accepts.

Basically, I'm trying to keep the interface simpler to use/understand.  A SENDONLY flag clearly specifies what it does.  Saying 'FULLMEMBER' requires someone to go read a 1500 page spec and figure out what exactly that means on IB.  And I have no idea what that would mean for UDP if iWarp adapters ever decide to support UDP.